packages/gatsby/src/redux/actions/public.js

```diff
@@ -481,6 +482,39 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
       deleteActions = findChildren(oldNode.children)
         .map(getNode)
         .map(createDeleteAction)
+
+      // Increment the node's contentDigest if it's set.
+      const oldNodeDigestId = parseInt(
+        oldNode.internal.contentDigest.split(`///`)[1]
+      )
+      if (Math.random() > 0.995) {
+        console.log({ oldNodeDigestId })
+      }
+      if (_.isNumber(oldNodeDigestId)) {
+        newDigestId = oldNodeDigestId + 1
+      }
+    }
+
+    node.internal.contentDigest = `${internalPage.path}-${internalPage.mode}-${internalPage.matchPath}-${internalPage.componentChunkName}///${newDigestId}`
```

Comment:
Hashing objects is expensive — particularly if the page object has a large context. Instead we'll construct the `contentDigest` by concatenating the critical info in the page object.

Creating the contentDigest was the most expensive operation before (with a large context) — about 20% of CPU time during my profile. And completely goes away with this.